test/unit_test/static/math_test.cpp

```diff
@@ -40,6 +40,11 @@ template<typename T1, typename T2, typename... Ts>
 
 #if __cpp_lib_constexpr_cmath || MP_UNITS_COMP_GCC
 
+static_assert(compare(fma(2 * m, 3 * m, 1 * m2), 7 * m2));
+static_assert(compare(fma(2.0 * s, 3.0 * Hz, 1.0), 7.0));
+static_assert(compare(fma(2 * s, 3 * Hz, 1), 7));
+static_assert(compare(fma(2.0, 3.0*m, 1.0*m), 7*m);
+static_assert(compare(fma(2.0*m, 3.0, 1.0*m), 7*m));
```

Comment:
Please, see <https://cpp1.godbolt.org/z/5xrq836dd>.
- Numbers should be multiplied by `* one` to make them a dimensionless quantity.
- With that done, a test case is a duplicate, so I removed it.
- According to the constraints on `std::fma`, arguments should be floating-point.
```suggestion
static_assert(compare(fma(2.0 * m, 3.0 * m, 1.0 * m2), 7.0 * m2));
static_assert(compare(fma(2.0 * s, 3.0 * Hz, 1.0 * one), 7.0 * one));
static_assert(compare(fma(2.0 * one, 3.0 * m, 1.0 * m), 7.0 * m));
static_assert(compare(fma(2.0 * m, 3.0 * one, 1.0 * m), 7.0 * m));
```

Reply:
That would be inconsistent with the design of the library.

Reply:
When I converted the integers to floating-point by adding `.0`,
and added `* one` to make them dimensionless quantities,
some tests became repeated, so I removed those.

Reply:
You're right, it works.
We can just restore the tests that used integers.
The implementation of `mp_units::fma` doesn't need to change: <https://cpp1.godbolt.org/z/d63K8YY6h>.